### DISCUSSION TITLE
How can I create a grid with 4 columns but if there are only 3; they take the full width?

### DISCUSSION THREAD

**Discussion by maxpaleo** [tailwindlabs/tailwindcss#13359] [Help]
Hi guys, 

This is probably a a ridiculous question; but for the life of me; I can't figure it out! 

I have been struggling with creating a simple grid layout with Tailwind, where the maximum number of columns in a row should be 4 if there are 4 columns, but there should be 3 columns if there are only 3. 

I am from Bootstrap, which is pretty straightforward, but I can't seem to get it right with TW. 

Note: I am trying to avoid using JS for this as well. 

Summary: 

- The grid needs a layout with a maximum of 4 columns. 
- If there are 5 columns, it should display 4 in one row and 1 in the second. (I figured out this part)
-If there are 3 columns, the columns should expand to fill the width of the row. 
-if there is just 1 column, it takes the full width. 

<img width="1710" alt="Screenshot 2024-03-25 at 7 15 08 PM" src="https://github.com/tailwindlabs/tailwindcss/assets/91980500/e4e4945b-2d52-475d-9251-882ecc665046">

<img width="1729" alt="Screenshot 2024-03-25 at 7 15 56 PM" src="https://github.com/tailwindlabs/tailwindcss/assets/91980500/9bea3aad-4a3c-47bb-8682-ae63aeabefeb">

**Comment by alex-krasikau** [comment]
Hey @maxpaleo!

Here's the solution: 
```css
grid grid-cols-[repeat(auto-fit,minmax(calc((100%-16px*3)/4),1fr))] gap-4
```

In this code, `16px` is the column gap size. It ensures each column has a minimum width equal to one card in a 4-column layout. The calculation `(100%-16px*3)/4` determines the card width in a 4-column layout (removes the gaps and divides by 4). Any extra space will be evenly distributed among the cards.

You can view a complete example here: [Tailwind Play Example](https://play.tailwindcss.com/qJodHv4oVd).

Please let me know if you have any questions.

<img width="1669" alt="CleanShot 2024-03-26 at 12 27 24@2x" src="https://github.com/tailwindlabs/tailwindcss/assets/97882909/05d8d81a-f967-46c6-aa96-7ef7436551be">
